javaslang/src/main/java/javaslang/collection/JavaConverters.java

```diff
@@ -0,0 +1,433 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _______
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  //  /\__\   JΛVΛSLΛNG
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/ \ /__\ \   Copyright 2014-2016 Javaslang, http://javaslang.io
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/\__\_/  \_//  \__/\_____/   Licensed under the Apache License, Version 2.0
+ */
+package javaslang.collection;
+
+import java.io.Serializable;
+import java.util.*;
+import java.util.Iterator;
+import java.util.List;
+
+import static javaslang.API.TODO;
+
+/**
+ * THIS CLASS IS INTENDED TO BE USED INTERNALLY ONLY!
+ * <p>
+ * This helper class provides methods that return views on Java collections.
+ * The view creation and back conversion take O(1).
+ *
+ * @author Daniel Dietrich, Pap Lőrinc, Sean Flanigan
+ * @since 2.1.0
+ */
+class JavaConverters {
+
+    private JavaConverters() {
+    }
+
+    static <T, C extends Seq<T>> ListView<T, C> asJava(C seq, ChangePolicy changePolicy) {
```

Comment:
Added exact type parameter C and returning the view type in order to be able to pull out the original delegate later from within a Javaslang collection.